spectralcluster/refinement.py

```diff
@@ -73,11 +73,11 @@ def __init__(self, p_percentile=0.95, thresholding_soft_multiplier=0.01):
     def refine(self, X):
         self.check_input(X)
         Y = np.copy(X)
-        for r in range(Y.shape[0]):
-            row_max = Y[r, :].max()
-            for c in range(Y.shape[1]):
-                if Y[r, c] < row_max * self.p_percentile:
-                    Y[r, c] *= self.multiplier
+        row_max = Y.max(axis=1)
+        row_max = np.expand_dims(row_max, axis=1)
+        is_smaller_or_not = Y < row_max * self.p_percentile
```

Comment:
Can you change to:
`is_smaller = Y < (row_max * self.p_percentile)`

The `_or_not` is unnecessary and confusing. The parenthesis enhances readability.

Reply:
You're right.